### ISSUE TITLE
Rationale of small inference batch size

### ISSUE THREAD

**Issue by cliangyu** [acl21/init-pools-dal#4]
`train_dataloader = data_obj.getSequentialDataLoader(indexes=trainSet, batch_size=5, data=simclr_data)
`
I noticed this line of code in 3.SimCLR_Losses.ipynb
May I know if there is any rationale of this fixed small batch size? What will happen if we increase this value? Because currently it takes 10 minutes per 50k images in CIFAR-10 which is relatively long.

**Comment by acl21** [comment]
In the case of the SimCLR method of initial pool sampling (see 5.2 in the paper for clarity), we take the average loss of the trained model on 4 SimCLR augmentations of an image (horizontal flipping, Gaussian blur, colour jitter, and grayscale). This average value is assigned to the image. This is why the `simclr_data` is sequential and can be split into sets of 5 (original image +  4 augmentations). Hence there are 50K images and not 10K (original initial pool size). 

I hope this explanation clears your doubt. :)

**Comment by cliangyu** [comment]
Great, have you ever tried tuning the hyperparameter of augmentation duplicates? say, if we have mroe augmentations.

**Comment by acl21** [comment]
No. I have simply taken the 4 easy augmentations that are popularly used in SimCLR training and stuck with them. It can be tuned but I did not tune at all.

**Comment by cliangyu** [comment]
I see, thanks for your insight!
